sky/tools/release_engine.py

```diff
@@ -114,8 +134,32 @@ def main():
     with open(sky_engine_revision_file, 'w') as stream:
         stream.write(commit_hash)
 
-    upload_artifacts(android_dist_root, 'android-arm', commit_hash)
-    upload_artifacts(linux_dist_root, 'linux-x64', commit_hash)
+    configs = [('android-arm', android_out_root),
+               ('linux-x64', linux_out_root)]
+
+    # Check for missing artifact files
+    missing_artifacts = []
+    for config, config_root in configs:
+        missing_artifacts.extend(find_missing_artifacts(config, config_root))
+    if missing_artifacts:
+        print ('Build is missing files:\n%s' %
+               '\n'.join('\t%s' % path for path in missing_artifacts))
+        return 1
+
+    # Build a zip file of artifacts
+    artifact_zip_path = 'out/artifacts.zip'
+    artifact_zip = zipfile.ZipFile(artifact_zip_path, 'w')
+    for config, config_root in configs:
+        for artifact_path in ZIP_ARTIFACTS[config]:
+            artifact_zip.write(os.path.join(config_root, artifact_path),
+                               os.path.join(config, os.path.basename(artifact_path)))
```

Comment:
The default is no compression.  Changed this to choose compression or not based on the file extension.